packages/core/src/VNodes/ChildNode.ts

```diff
@@ -0,0 +1,484 @@
+import { Constructor, nodeLength } from '../../../utils/src/utils';
+import { VNode, RelativePosition, isLeaf, Predicate, Typeguard } from './VNode';
+
+export abstract class ChildNode extends VNode {
```

Comment:
I don't really get the added value of separating these methods from other `VNode` methods. Plus, why is it called `ChildNode`? It's confusing.

Reply:
I want VNode to be the lightest possible. In my mind, VNode is an interface. It is not in practice for the reasons I explained in the commit message, but that answers your question since interfaces cannot have method implementations. I'm open for name suggestions though.